Kosmos. Copyright (c) 2026 Diego Cibils. MIT; see LICENSE. - runtime/patches/netsurf/README.md says why.
--- a/libdom/src/events/event_target.c	2026-09-30 15:00:35.187435958 -0300
+++ b/libdom/src/events/event_target.c	2026-09-30 15:00:35.203512865 -0300
@@ -19,8 +19,14 @@
 #include "utils/utils.h"
 #include "utils/validate.h"
 
+/* Kosmos: how many listeners exist, anywhere in this process. While there
+ * are none, dispatch.c does not make the mutation events nobody would hear
+ * (runtime/patches/netsurf/README.md). */
+unsigned long _dom_event_listeners_live = 0;
+
 static void event_target_destroy_listener(struct listener_entry *e)
 {
+	_dom_event_listeners_live--;
 	list_del(&e->list);
 	dom_event_listener_unref(e->listener);
 	dom_string_unref(e->type);
@@ -93,6 +99,8 @@
 		list_append(&eti->listeners->list, &le->list);
 	}
 
+	_dom_event_listeners_live++;
+
 	return DOM_NO_ERR;
 }
 
@@ -137,6 +145,7 @@
 				dom_event_listener_unref(le->listener);
 				dom_string_unref(le->type);
 				free(le);
+				_dom_event_listeners_live--;
 				break;
 			}
 
